app/assets/javascripts/discourse/app/components/group-manage-save-button.gjs

```diff
@@ -13,44 +13,29 @@ import { i18n } from "discourse-i18n";
 export default class GroupManageSaveButton extends Component {
   @service modal;
   @service groupAutomaticMembersDialog;
+  @service router;
 
   saving = null;
   disabled = false;
   updateExistingUsers = null;
 
-  @computedOr("model.flair_icon", "model.flair_url") hasFlair;
-
   @discourseComputed("saving")
   savingText(saving) {
     return saving ? i18n("saving") : i18n("save");
   }
 
-  @discourseComputed(
-    "model.visibility_level",
-    "model.primary_group",
-    "hasFlair"
-  )
-  privateGroupNameNotice(visibilityLevel, isPrimaryGroup, hasFlair) {
-    if (visibilityLevel === 0) {
-      return;
-    }
-
-    if (isPrimaryGroup) {
-      return i18n("admin.groups.manage.alert.primary_group", {
-        group_name: this.model.name,
-      });
-    } else if (hasFlair) {
-      return i18n("admin.groups.manage.alert.flair_group", {
-        group_name: this.model.name,
-      });
-    }
-  }
-
   @action
   setUpdateExistingUsers(value) {
     this.updateExistingUsers = value;
   }
 
+  get shouldRenderWarningFlair() {
+    if (this.router.currentRoute.name === "group.manage.membership") {
+      return false;
+    }
+    return true;
```

Comment:
```suggestion
    return this.router.currentRouteName !== "group.manage.membership";
```

* `this.router.currentRoute.name` => `this.router.currentRouteName` (see [public API](https://api.emberjs.com/ember/release/classes/RouterService#currentRouteName)) - accessing `currentRouteName` is faster, while `currentRoute.name` requires property access traversal (accessing `currentRoute` object first, then its `name` property)
* the direct return with boolean expression is faster. In practice, the difference is tiny (nanoseconds), but the direct return will consistently benchmark faster because it avoids the branching overhead. The if-clause version requires comparison evaluation, branch prediction, and two separate return statements (different code paths)